include/zcbor_diag_print.h

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef ZCBOR_DIAG_PRINT_H__
+#define ZCBOR_DIAG_PRINT_H__
+
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef ZCBOR_DIAG_PRINT_FUNC
+#include <zephyr/sys/printk.h>
+#define zcbor_do_diag_print(...) printk(__VA_ARGS__)
+#else
+#define zcbor_do_diag_print(...) ZCBOR_DIAG_PRINT_FUNC(__VA_ARGS__)
+#endif
+
+static bool indent_printed = false;
+
+__attribute__((used))
+static void zcbor_print_indent(uint8_t indent_len)
```

Comment:
nit: I guess you want the wrapping functionality of `indent_len`? Otherwise why not just use a full uint32.